pkg/clusterrouter/proxy/proxy_server.go

```diff
@@ -220,7 +222,20 @@ func (r *RemoteService) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	req.URL.Host = req.Host
-	if r.auth == nil {
+	transport, err := r.getTransport()
+	if err != nil {
+		er.Error(rw, req, err)
+		return
+	}
+
+	if r.cluster.Status.Driver == "googleKubernetesEngine" && r.cluster.Spec.GenericEngineConfig != nil {
+		cred, _ := (*r.cluster.Spec.GenericEngineConfig)["credential"].(string)
+		transport, err = gke.Oauth2Transport(context.Background(), transport, cred)
+		if err != nil {
+			er.Error(rw, req, fmt.Errorf("unable to retrieve token source for GKE oauth2: %v", err))
+			return
+		}
+	} else if r.auth == nil {
```

Comment:
Is there a reason we should have this not run when the cluster is GKE ? 
I don't get why we didn't leave the `getTransport` call where it was and put this new code below that. Do we always want Oauth to skip the `auth()` calls here ?

Reply:
It is to emphasize that we are NOT using the bearer token. The transport is handling it for us.

Reply:
Assuming you don't have to delete the `Authorization` header because it never existed ? Ok.